destination/keepsafe.go

```diff
@@ -64,8 +65,6 @@ func (k *keepSafe) GetAll() [][]byte {
 }
 
 func (k *keepSafe) Stop() {
-	k.Lock()
-	close(k.closed)
+	k.closed = true
```

Comment:
is it safe to do un-synchronized writes and reads on the same variable? i believe with go's memory model, the answer is no.
i would keep the lock around the write (or if you want to be really fancy, do an atomic operation on an int, but that seems overkill)

Reply:
Yeah, it's not safe heh. I was thinking atomic. We can't keep the write locks around it, that is what is causing the deadlock. The other goroutine will never acquire the lock, because goroutine 1 won't release it, because it is waiting for gourtine 2 to return and decrement the waitgroup.